### PR TITLE
[Develop] Use names for tutorial URL ids

### PR DESCRIPTION
**Pull request by rschamp**
To make URLs more friendly, use e.g., `tutorial=getting-started` instead of `tutorial=1`.  This will also help with some URL handling we're doing for the new project page.

### CODE REVIEW

File: src/lib/tutorial-from-url.js [hunk 1]
```diff
@@ -36,15 +36,10 @@ const getDeckIdFromUrlId = urlId => {
...
-            queryParams.tutorial
-    );
-    if (!isNaN(tutorialID)) {
-        return getDeckIdFromUrlId(tutorialID);
-    }
-    return null;
+    const tutorialID = Array.isArray(queryParams.tutorial) ?
+        queryParams.tutorial[0] :
+        queryParams.tutorial;
+    return getDeckIdFromUrlId(tutorialID);
```
Comment (kchadha) [inline src/lib/tutorial-from-url.js]: Should we also add an early exit in getDeckIdFromUrlId if tutorialID is `undefined`...?

File: src/lib/libraries/decks/index.jsx [hunk 2]
```diff
@@ -423,7 +423,7 @@ export default {
             ]
         }
         ],
-        urlId: 2
+        urlId: 'animate-a-name'
```
Comment (chrisgarrity) [inline src/lib/libraries/decks/index.jsx]: Currently: `projects/editor/?tip_bar=name`

File: src/lib/tutorial-from-url.js (hunk 1 above)
Comment (kchadha) [reply src/lib/tutorial-from-url.js]: Would a tutorial that is missing the 'urlId' field get matched to this tutorialId in this case?

File: src/lib/libraries/decks/index.jsx [hunk 3]
```diff
@@ -492,7 +492,7 @@ export default {
             ]
         }
         ],
-        urlId: 3
+        urlId: 'make-music'
```
Comment (chrisgarrity) [inline src/lib/libraries/decks/index.jsx]: `/projects/editor/?tip_bar=music`

File: src/lib/tutorial-from-url.js (hunk 1 above)
Comment (rschamp) [reply src/lib/tutorial-from-url.js]: Good catch, we should exit early!

**Comment by rschamp** [comment]
@chrisgarrity I found one more, `hide`. But none of the rest seem to map, so I left them alone. Happy to take any other suggestions.

@kchadha I fixed that bug and added a regression test.

**Comment by thisandagain** [comment]
This looks good. Thanks @rschamp !

**Comment by chrisgarrity** [comment]
👍

**Pull request merged**